src/Datasource/EntityInterface.php

```diff
@@ -61,6 +61,30 @@ public function setVirtual(array $fields, bool $merge = false);
      */
     public function getVirtual(): array;
 
+    /**
+     * Returns whether a field is an original one
+     *
+     * @return bool
+     */
+    public function isOriginalField(string $name): bool;
+
+    /**
+     * Returns an array of field names previously set using `setOriginalField()`
+     * The entity was initialized with
+     *
+     * @return array<string>
+     */
+    public function getOriginalFields(): array;
+
+    /**
+     * Sets the given field or a list of fields to set as original
+     *
+     * @param array<string>|string $field the name of a field or a list of fields to set as original
+     * @param bool $merge
+     * @return $this
+     */
+    public function setOriginalField(string|array $field, bool $merge = true);
```

Comment:
Do we need this method? Wouldn't saving the original value be an internal only operation when setting a field value or patching the entity?

Reply:
I also thought this seems to be a bit weird but then I saw the needed changes in the Marshaller which require this method to be public.
Maybe @fabian-mcfly you can explain why this needs to be a public method?

Reply:
The Marshaller creates a new instance without any parameters:
https://github.com/cakephp/cakephp/blob/f104f2da8071652136d270ceac9b469eb4f42f79/src/ORM/Marshaller.php#L180

After that, the list of properties is built, leaving fields with errors out in case the entity implements `InvalidPropertyInterface`:
https://github.com/cakephp/cakephp/blob/f104f2da8071652136d270ceac9b469eb4f42f79/src/ORM/Marshaller.php#L197-L202
I didn't want to touch too much code there, but I can change it so the entity will be created after the property list is available. This way, the method does not need to be public in `Entity`, and can be left out of `EntityInterface`.

Reply:
Would it be possible to create this kind of functionality without making that method publicly accessible?